python/hail/api2/matrixtable.py

```diff
@@ -2222,5 +2222,118 @@ def unpersist(self):
         return self._jvds.same(other._jvds, tolerance)
-        
+
+    @handle_py4j
+    @typecheck(datasets=matrix_table_type)
+    def union_rows(*datasets):
+        """Take the union of dataset rows.
+
+        Examples
+        --------
+
+        .. testsetup::
+
+            dataset_to_union = dataset
+            dataset_to_union_2 = dataset
+
+        Union the rows of two datasets:
+
+        >>> dataset_result = dataset.union_rows(dataset_to_union)
+
+        Given a list of datasets, take the union of all rows:
+
+        >>> all_datasets = [dataset, dataset_to_union, dataset_to_union_2]
+
+        The following three syntaxes are equivalent:
+
+        >>> dataset_result = dataset.union_rows(dataset_to_union, dataset_to_union_2)
+        >>> dataset_result = all_datasets[0].union_rows(*all_datasets[1:])
+        >>> dataset_result = MatrixTable.union_rows(*all_datsets)
+
+        Notes
+        -----
+
+        In order to combine two datasets, four requirements must be met:
+
+         - The column keys must be identical (in both type and equality, field
+           order within structs matters).
+         - The row key schema and row field schemas must match.
+         - The entry schemas must match.
+
+        The column fields in the resulting dataset are simply the column
+        annotations from the first dataset; the column field schemas do not
+        need to match.
+
+        This method does not deduplicate; if a row exists identically in two
+        datasets, then it will be duplicated in the result.
+
+        Warning
+        -------
+        This method can trigger a shuffle, if partitions from two datasets
+        overlap.
+
+        Parameters
+        ----------
+        datasets : varargs of :class:`MatrixTable`
+            Datasets to combine..
+
+        Returns
+        -------
+        :class:`MatrixTable`
+            Dataset with rows from each member of `datasets`.
+        """
+        if len(datasets) == 0:
+            raise ValueError('Expected at least one argument')
+        elif len(datasets) == 1:
+            return datasets[0]
+        else:
+            return MatrixTable(Env.hail().variant.MatrixTable.union_rows([d._jvds for d in datasets]))
+
+    @handle_py4j
+    @typecheck_method(other=matrix_table_type)
+    def union_cols(self, other):
+        """Take the union of dataset columns.
+
+        Examples
+        --------
+
+        .. testsetup::
+
+            dataset_to_union = dataset
+
+        Union the columns of two datasets:
+
+        >>> dataset_result = dataset.union_cols(dataset_to_union)
+
+        Notes
+        -----
+
+        In order to combine two datasets, four requirements must be met:
+
+         - The column key schemas must match.
+         - The column field schemas must match.
+         - The row key schemas must match.
+         - The entry schemas must match.
+
+        The row fields in the resulting dataset are the row fields from the
```

Comment:
It looks like row fields is the same as row annotations from context here. I find row fields to be confusing because I think it's referring to the row key schema. I can't think of a better name write now. If we stick with this naming scheme, we need to make the overview page very clear (the one with the VDS diagram) -- probably need to redo that anyways.